From 196d6bb332c05462c9a9d913a0fa44699db95344 Mon Sep 17 00:00:00 2001
From: Jean Christophe Beyler <jean.christophe.beyler@intel.com>
Date: Mon, 11 Nov 2013 16:53:53 -0800
Subject: Dalvik: Insert Pass Before Does Not Insert Correctly

BZ: 151255

The insert pass before for a user plugin is wrong. It does not set the new
pass as a next pass, thus the insertion being thought correct actually does
not work.

Change-Id: I0209abcd1b3acdd66263367936c7a054c3f73df1
Category: device enablement
Domain: AOSP.Dalvik-Compiler-ME
Origin: internal
Upstream-Candidate: no, need rework
Signed-off-by: Jean Christophe Beyler <jean.christophe.beyler@intel.com>
Signed-off-by: Ivan Popov <ivan.g.popov@intel.com>
---
 vm/compiler/PassDriver.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/vm/compiler/PassDriver.cpp b/vm/compiler/PassDriver.cpp
index ea6172f..42dca34 100644
--- a/vm/compiler/PassDriver.cpp
+++ b/vm/compiler/PassDriver.cpp
@@ -664,7 +664,7 @@ bool dvmCompilerInsertUserPass (Pass *newPass, const char *name, enum PassInstru
                 }
                 else
                 {
-                    previous->setNext (curPass);
+                    previous->setNext (newPass);
                 }
 
                 curPass->setPrevious (newPass);
-- 
1.7.4.1

